src/gt4py/next/ffront/foast_to_itir.py

```diff
@@ -61,15 +61,27 @@ class FieldOperatorLowering(NodeTranslator):
     <class 'gt4py.next.iterator.ir.FunctionDefinition'>
     >>> lowered.id
     SymbolName('fieldop')
-    >>> lowered.params
-    [Sym(id=SymbolName('inp'), kind='Iterator', dtype=('float64', False))]
+    >>> lowered.params # doctest: +ELLIPSIS
+    [Sym(location=..., id=SymbolName('inp'), kind='Iterator', dtype=('float64', False))]
     """
 
     uid_generator: UIDGenerator = dataclasses.field(default_factory=UIDGenerator)
+    preserve_location: bool = True
 
     @classmethod
-    def apply(cls, node: foast.LocatedNode) -> itir.Expr:
-        return cls().visit(node)
+    def apply(cls, node: foast.LocatedNode, preserve_location: bool = True) -> itir.Expr:
+        return cls(preserve_location=preserve_location).visit(node)
+
+    def visit(self, node: concepts.RootNode, **kwargs: extended_typing.Any) -> extended_typing.Any:
+        result = super().visit(node, **kwargs)
+        if (
+            hasattr(node, "location")
+            and hasattr(result, "location")
+            and not isinstance(node, foast.Name)
+            and self.preserve_location
```

Comment:
Why is the location preservation disabled for `foast.Name` nodes?
```suggestion
            and not isinstance(node, foast.Name)
            and self.preserve_location
```
If this wouldn't be needed the `preserve_location` flag could easily be added to `PreserveLocation` class making this custom handling unneeded.